File: research/v6diff.py
import json, sys, manzil_v6 as v
cases = json.load(open(sys.argv[1]))
out = []
for c in cases:
    C = v.Cards(own_all=True)
    r = v.play_board(C, list(c["you"]), list(c["sky"]), c["tonight"],
                     leader=c["leader"], you_depth=c["youDepth"], depth=c["depth"])
    out.append({"win": r["win"], "you": r["you"], "sky": r["sky"], "flips": r["flips"]})
print(json.dumps(out))

File: research/v6diff.js
const A = require("./manzil-engine-v6-AUG24.js");
const cases = JSON.parse(require("fs").readFileSync(process.argv[2], "utf8"));
const out = cases.map(c => {
  const C = A.makeCards({ lvl: 2 });
  const r = A.playBoard({ C, tonight: c.tonight, you: c.you, sky: c.sky,
                          leader: c.leader, depth: c.depth, youDepth: c.youDepth });
  return { win: r.winner, you: r.you, sky: r.sky, flips: r.flips };
});
console.log(JSON.stringify(out));
